internal/workflow/processing.go

```diff
@@ -911,18 +916,43 @@ func (w *ProcessingWorkflow) preprocessing(ctx temporalsdk_workflow.Context, tin
 		WorkflowID:        fmt.Sprintf("%s-%s", w.cfg.Preprocessing.Temporal.WorkflowName, uuid.New().String()),
 		ParentClosePolicy: temporalapi_enums.PARENT_CLOSE_POLICY_TERMINATE,
 	})
-	var result preprocessing.WorkflowResult
+	var ppResult preprocessing.WorkflowResult
 	err = temporalsdk_workflow.ExecuteChildWorkflow(
 		preCtx,
 		w.cfg.Preprocessing.Temporal.WorkflowName,
 		preprocessing.WorkflowParams{RelativePath: relPath},
-	).Get(preCtx, &result)
+	).Get(preCtx, &ppResult)
 	if err != nil {
 		return err
 	}
 
-	tinfo.TempPath = filepath.Join(w.cfg.Preprocessing.SharedPath, filepath.Clean(result.RelativePath))
+	tinfo.TempPath = filepath.Join(w.cfg.Preprocessing.SharedPath, filepath.Clean(ppResult.RelativePath))
 	tinfo.IsDir = true
 
-	return nil
+	opts := withLocalActivityOpts(ctx)
+	var savePPTasksResult localact.SavePreprocessingTasksActivityResult
+	err = temporalsdk_workflow.ExecuteLocalActivity(
+		opts,
+		localact.SavePreprocessingTasksActivity,
+		localact.SavePreprocessingTasksActivityParams{
+			PkgSvc:               w.pkgsvc,
+			RNG:                  w.rng,
+			PreservationActionID: tinfo.PreservationActionID,
+			Tasks:                ppResult.PreservationTasks,
+		},
+	).Get(opts, &savePPTasksResult)
+	if err != nil {
+		return err
+	}
```

Comment:
We could check if there are PreservationTasks in the preprocessing result and only execute the activity if needed.

Reply:
Yeah, good idea. :+1: